server/src/main/java/org/opensearch/index/query/NestedQueryBuilder.java

```diff
@@ -505,4 +506,12 @@ public TopDocsAndMaxScore topDocs(SearchHit hit) throws IOException {
             }
         }
     }
+
+    @Override
+    public void visit(QueryBuilderVisitor visitor) {
+        visitor.accept(this);
+        if (query != null) {
```

Comment:
Why is the check for query not being null not present in previous PRs for other query clause builders? Is this a new scenario, or is the inner nested query different from other queries? If the latter is true, please add a comment to the code specifying why we need this check for nested

Reply:
We have kept it there for 2 reasons
1. To pass unit tests
2. The same is been followed in other query builders https://github.com/opensearch-project/OpenSearch/pull/10110/files#diff-056910f9008c16e48b353a045b19d604464d8dbf03992cda29e128af92afec8c

Reply:
reg 1 - what exact test do you mean, one that has been added in the PR or something from existing codebase?
reg 2 - for span query those `exclude` and `include` sub-queries are special case, if something like that is for nested better have a comment, this is small PR with only one query, easier to do comparing to initial PR where you've  touched multiple queries